### PR TITLE
Search protocol over jrt + protobuf (take 2) 

### PR DESCRIPTION
**Pull request by ollivir**
This time `searchlib` is moved from `container-disc` to `container-search-and-docproc` and `protobuf-java` is included there. The change in `VespaModel` is to avoid the `StandaloneContainerApplication` construction code path needlessly trying to process rank profiles which would try to load `searchlib` classes.

FYI @gjoranv

### CODE REVIEW

**Review by bratseth — APPROVED** [review]

**Comment by ollivir** [comment]
ping merge stop

File: container-dependency-versions/pom.xml [hunk 1]
```diff
@@ -383,6 +383,11 @@
                 <artifactId>xml-apis</artifactId>
                 <version>1.4.01</version>
             </dependency>
+            <dependency>
```
Comment (gjoranv) [inline container-dependency-versions/pom.xml]: This must be moved to parent/pom.xml. This pom is only for dependencies that are exposed by the container to bundles.

File: container-search/pom.xml [hunk 2]
```diff
@@ -138,6 +138,11 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>com.google.protobuf</groupId>
+      <artifactId>protobuf-java</artifactId>
+      <scope>provided</scope>
```
Comment (gjoranv) [inline container-search/pom.xml]: Couldn't this be compile, and then you could skip this dep in container-search-and-docproc?

File: container-dependency-versions/pom.xml (hunk 1 above)
Comment (ollivir) [reply container-dependency-versions/pom.xml]: Moved in https://github.com/vespa-engine/vespa/pull/8879 along with the compile scope change.

**Pull request merged**